src/core/a-entity.js

```diff
@@ -346,16 +346,32 @@ var proto = Object.create(ANode.prototype, {
     writable: window.debug
   },
 
+  /**
+   * Initialize dependencies of a component.
+   *
+   * @param {string} name - Root component name.
+   */
   initComponentDependencies: {
     value: function (name) {
       var self = this;
       var component = COMPONENTS[name];
       var dependencies;
+
+      // Not a component.
       if (!component) { return; }
+
+      // No dependencies.
       dependencies = COMPONENTS[name].dependencies;
       if (!dependencies) { return; }
-      dependencies.forEach(function (component) {
-        self.initComponent(component, undefined, true);
+
+      // Initialize dependencies.
+      dependencies.forEach(function initializeDependency (componentName) {
+        // Call getAttribute to initialize the data from the DOM.
+        self.initComponent(
+          componentName,
+          HTMLElement.prototype.getAttribute.call(self, componentName),
```

Comment:
At some point (I cannot longer tell you if the case) a null value was interpreted as a component being removed. This is why `undefined` value was passed before.